lib/ClangImporter/ClangImporter.cpp

```diff
@@ -3482,16 +3472,16 @@ void ClangModuleUnit::getImportedModules(
   for (auto importMod : imported) {
     auto wrapper = owner.getWrapperForModule(importMod);
 
-    auto actualMod = wrapper->getOverlayModule();
-    if (!actualMod) {
+    auto importTopLevel = importMod->getTopLevelModule();
+    auto actualMod = getASTContext().getModuleByName(importTopLevel->Name);
+    if (!actualMod || actualMod->isNonSwiftModule()) {
```

Comment:
The `lookupValue` function that this code is in, is *way* too long. It would be great if you could at least constify every local to the extent possible, and/or break it up. Not your fault, just a chance to tidy up a (IMO) mess.